### ISSUE TITLE
stochastic test failure of `StationaryBlockBootstrap`

### ISSUE THREAD

**Issue by fkiraly** [astrogilda/tsbootstrap#73]
`StationaryBlockBootstrap` causes failures as follows:

* in `TestAllBootstraps`, stochastic failure in `test_input_output_contract`
* in #66, same stochastic failure in `test_base_block_bootstrap`

The failure occurs when testing `StationaryBlockBootstrap` (or `BaseBlockBootstrap` when dispatching to that class via `"stationary"`) with its `BlockGenerator.generate_overlapping_blocks` generating empty blocks, causing the `validate_block_indices` at the end to trip. This is due to a bug in `StationaryBlockBootstrap` that causes sampling of an empty list of blocks.

Internally, the condition `if block_length < self.min_block_length`: interrupts a while loop, and in joint investigation today with @astrogilda and @benHeid, it seemed that this should not be `break` but `continue` - but if we change to `continue`, the loops does not terminate as `_get_total_length_covered` returns 0.

The break condition `if block_length < self.min_block_length` is reached since `min_block_length` is set to 2, despite the constructor getting a `None`, as it copies the `avg_block_length` from `BlockSampler`, when `None` is written to `self.min_block_length`, the replace happens through `property` setter magic.

Ultimately, we have decided that the failure is likely due to a bug, and the refactor may have caused it to appear with a failure, rather than directly cause it. For now, the test is hence skipped - and the issue may become obsoltee if, say, API rework steps from #70 are followed re `BaseBlockBootstrap`.

To unskip the test and check any fix:

* remove `StationaryBlockBootstrap` from `TEMPORARY_EXCLUDED_OBJECTS`
* unskip `test_base_block_bootstrap` (if the test is still present)

**Comment by fkiraly** [comment]
```python
import numpy as np
from tsbootstrap import StationaryBlockBootstrap

n_bootstraps=1
rng=None
X=np.array([[0., 0.],
       [0., 0.],
       [0., 0.],
       [0., 0.],
       [0., 0.],
       [0., 0.],
       [0., 0.],
       [0., 0.],
       [0., 0.]])

block_length = np.random.randint(1, int(0.8 * X.shape[0]) - 1)
bootstrap = StationaryBlockBootstrap(
    block_length=block_length,
    n_bootstraps=n_bootstraps,
    rng=rng,
)

list(bootstrap.bootstrap(X))
```

**Comment by fkiraly** [comment]
```python
from tsbootstrap import StationaryBlockBootstrap
from tsbootstrap.utils import check_estimator

check_estimator(StationaryBlockBootstrap, raise_exceptions=True)
```

**Comment by astrogilda** [comment]
#82 is a repeat of this. both issues fixed by pr #103
